lib/core/feature-flags/src/lib/services/debug-features.service.ts

```diff
@@ -0,0 +1,91 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Inject, Injectable, Optional } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { skip, switchMap } from 'rxjs/operators';
+import {
+    IDebugFeaturesService,
+    IFeaturesService,
+    FlagChangeset,
+    OverridableFeaturesServiceToken,
+    WritableFeaturesServiceToken,
+    WritableFeaturesServiceConfigToken,
+    WritableFeaturesServiceConfig,
+    FlagSet,
+    IWritableFeaturesService
+} from '../interfaces/features.interface';
+import { StorageService } from 'public-api';
+
+@Injectable()
+export class DebugFeaturesService implements IDebugFeaturesService {
+    private isInDebugMode: BehaviorSubject<boolean>;
+    private isInDebugMode$: Observable<boolean>;
+
+    get storageKey(): string {
+        return `${this.config?.storageKey || 'feature-flags'}-override`;
+    }
+
+    constructor(
+        @Inject(OverridableFeaturesServiceToken) private overriddenFeaturesService: IFeaturesService,
+        @Inject(WritableFeaturesServiceToken) private writableFeaturesService: IFeaturesService & IWritableFeaturesService,
+        private storageService: StorageService,
+        @Optional() @Inject(WritableFeaturesServiceConfigToken) private config?: WritableFeaturesServiceConfig
+    ) {
+        this.isInDebugMode = new BehaviorSubject<boolean>(JSON.parse(this.storageService.getItem(this.storageKey) || 'false'));
+        this.isInDebugMode$ = this.isInDebugMode.asObservable();
+
+        this.isInDebugMode.pipe(skip(1)).subscribe((debugMode) => {
+            this.storageService.setItem(this.storageKey, JSON.stringify(debugMode));
+        });
+    }
+
+    isOn$(key: string): Observable<boolean> {
+        return this.isInDebugMode$.pipe(
+            switchMap((isInDebugMode) => (isInDebugMode ? this.writableFeaturesService : this.overriddenFeaturesService).isOn$(key))
+        );
+    }
+
+    isOff$(key: string): Observable<boolean> {
+        return this.isInDebugMode$.pipe(
+            switchMap((isInDebugMode) => (isInDebugMode ? this.writableFeaturesService : this.overriddenFeaturesService).isOff$(key))
+        );
+    }
+
+    getFlags$(): Observable<FlagChangeset> {
+        return this.isInDebugMode$.pipe(
+            switchMap((isInDebugMode) => (isInDebugMode ? this.writableFeaturesService : this.overriddenFeaturesService).getFlags$())
+        );
+    }
+
+    getFlagsSnapshot(): FlagChangeset {
+        // eslint-disable-next-line rxjs/no-subject-value
```

Comment:
I presume this is the legacy/deprecated api used here?

Reply:
I'll check it and eventually mark it as deprecated. We can remove it after being sure the method is not used inside `hxp-frontend-apps` (spoiler: it's used)